spring-test/src/main/java/org/springframework/test/web/client/match/MockRestRequestMatchers.java

```diff
@@ -79,11 +79,13 @@ public static RequestMatcher requestTo(final Matcher<String> matcher) {
 	/**
 	 * Assert the request URI string.
 	 * @param expectedUri the expected URI
+	 * @param uriVars zero or more URI variables to populate the expected URI
 	 * @return the request matcher
 	 */
-	public static RequestMatcher requestTo(final String expectedUri) {
+	public static RequestMatcher requestTo(final String expectedUri, final Object... uriVars) {
 		Assert.notNull(expectedUri, "'uri' must not be null");
-		return request -> assertEquals("Request URI", expectedUri, request.getURI().toString());
+		String uri = UriComponentsBuilder.fromUriString(expectedUri).buildAndExpand(uriVars).encode().toString();
```

Comment:
Sounds good! Wasn't aware of the regression but I'll work on these changes later.